packages/api/src/controllers/webhook.ts

```diff
@@ -345,4 +279,93 @@ app.delete("/", authorizer({}), async (req, res) => {
   res.end();
 });
 
+const requestsFieldsMap: FieldsMap = {
+  id: `webhook_response.ID`,
+  createdAt: { val: `webhook_response.data->'createdAt'`, type: "int" },
+  userId: `webhook_response.data->>'userId'`,
+  event: `webhook_response.data->>'event'`,
+  statusCode: `webhook_response.data->'response'->>'status'`,
+};
+
+app.get("/:id/requests", authorizer({}), async (req, res) => {
+  let { limit, cursor, all, event, allUsers, order, filters, count } =
+    req.query;
+  if (isNaN(parseInt(limit))) {
+    limit = undefined;
+  }
+  if (!order) {
+    order = "createdAt-true";
+  }
+
+  if (req.user.admin && allUsers && allUsers !== "false") {
+    const query = parseFilters(requestsFieldsMap, filters);
+    query.push(sql`webhook_response.data->>'webhookId' = ${req.params.id}`);
+    if (!all || all === "false") {
+      query.push(sql`webhook_response.data->>'deleted' IS NULL`);
+    }
+
+    let fields =
+      " webhook_response.id as id, webhook_response.data as data, users.id as usersId, users.data as usersdata";
+    if (count) {
+      fields = fields + ", count(*) OVER() AS count";
+    }
+    const from = `webhook_response left join users on webhook_response.data->>'userId' = users.id`;
+    const [output, newCursor] = await db.webhookResponse.find(query, {
+      limit,
+      cursor,
+      fields,
+      from,
+      order: parseOrder(requestsFieldsMap, order),
+      process: ({ data, usersdata, count: c }) => {
+        if (count) {
+          res.set("X-Total-Count", c);
+        }
+        return { ...data, user: db.user.cleanWriteOnlyResponse(usersdata) };
+      },
+    });
+
+    res.status(200);
+
+    if (output.length > 0 && newCursor) {
+      res.links({ next: makeNextHREF(req, newCursor) });
+    }
+    return res.json(output);
+  }
+
+  const query = parseFilters(requestsFieldsMap, filters);
+  query.push(sql`webhook_response.data->>'userId' = ${req.user.id}`);
+  query.push(sql`webhook_response.data->>'webhookId' = ${req.params.id}`);
+
+  if (!all || all === "false") {
+    query.push(sql`webhook_response.data->>'deleted' IS NULL`);
+  }
```

Comment:
I believe this option shouldn't be available in non-admin queries. Lmk if you saw this in another handler 👀

Reply:
Huh, yeah good point! I think I must have copied this from here: https://github.com/livepeer/studio/blob/f71d79de9fccb99889f697d11b7a94af290e349d/packages/api/src/controllers/webhook.ts#L99 I'll fix it there too